src/parser/parser.cc

```diff
@@ -0,0 +1,1103 @@
+      return Type();
+    }
+  }
+
+  Attrs ParseAttrs(const std::string& type_key) {
+    Map<String, ObjectRef> kwargs;
+    auto attrs = tvm::ReflectionVTable::Global()->CreateObject(type_key, kwargs);
+    LOG(FATAL) << Attrs();
+    return Attrs();
+  }
+
+  Function ParseFunctionDef() {
+    PushScope();
+    PushTypeScope();
+
+    Array<TypeVar> generics;
+    if (Peek()->token_type == TokenType::LSquare) {
+      // If we have generics we need to add a type scope.
+      PushTypeScope();
+      generics = ParseSequence<TypeVar>(TokenType::LSquare, TokenType::Comma, TokenType::RSquare, [&]() {
+        auto type_var_name = Match(TokenType::Identifier).ToString();
+        return BindTypeVar(type_var_name, TypeKind::kType);
+      });
+    }
+
+    auto params = ParseSequence<Var>(TokenType::OpenParen, TokenType::Comma, TokenType::CloseParen, [&]() {
+      auto token = Match(TokenType::Local);
+      auto string = token.ToString();
+      Type type;
+      if (WhenMatch(TokenType::Colon)) {
+        type = ParseType();
+      }
+      return BindVar(string, type);
+    });
+
+    Type ret_type;
+    if (WhenMatch(TokenType::Minus)) {
+      Match(TokenType::RAngle);
+      ret_type = ParseType();
+    }
+
+    auto body = Block<Expr>([&]() {
+      return ParseExpr();
+    });
+
+    PopTypeScopes(1);
+    PopScopes(1);
+
+    return relay::Function(params, body, ret_type, generics);
+  }
+
+  Expr ParseIf() {
+    Consume(TokenType::If);
+    auto guard = Parens<Expr>([&] {
+      return ParseExpr();
+    });
+
+    auto true_branch = Block<Expr>([&] {
+      return ParseExpr();
+    });
+
+    Match(TokenType::Else);
+
+    auto false_branch = Block<Expr>([&] {
+      return ParseExpr();
+    });
+
+    return relay::If(guard, true_branch, false_branch);
+  }
+
+  Expr ParseMatch(bool is_partial) {
+    LOG(FATAL) << "parse match";
+  }
+
+  Expr ParseExpr() {
+    return ConsumeWhitespace<Expr>([this] {
+      std::vector<Expr> exprs;
+
+      while (true) {
+        auto next = Peek();
+        switch (next->token_type) {
+          // For graph or let, match first rhs, then invoke ParseBindingExpr
+          // ParseBindingExpression then parse_lhs() parse_rhs() ';' continue
+          case TokenType::Let:
+            exprs.push_back(ParseBindingExpr());
+            break;
+          case TokenType::Match:
+          case TokenType::PartialMatch:
+            bool is_partial = next->token_type == PartialMatch;
+            Consume(nest->token_type);
+            exprs.push_back(ParseMatch(is_partial));
+            break;
+          case TokenType::If: {
+            exprs.push_back(ParseIf());
+            break;
+          }
+          case TokenType::Graph:
+            if (Lookahead(2)->token_type == TokenType::Equal) {
+              exprs.push_back(ParseBindingExpr());
+              break;
+            }
+            // intentional fall through here.
+          default: {
+            DisplayNextN(100);
```

Comment:
name the magic number

Reply:
This is just an arbitrary call to debugging code that is there while I fix the last couple CI cases.

Reply:
i.e it won't be in there when I land